src/openms_gui/source/VISUAL/DIALOGS/ToolsDialog.cpp

```diff
@@ -83,37 +87,26 @@ namespace OpenMS
     main_grid->addWidget(label, 1, 0);
     QStringList list;
 
-    if (layer_type == LayerData::DT_PEAK)
-    {
-      list << "FileFilter" << "FileInfo"
-           << "NoiseFilterGaussian" << "NoiseFilterSGolay"
-           << "BaselineFilter" << "PeakPickerHiRes"
-           << "PeakPickerWavelet" << "Resampler"
-           << "MapNormalizer" << "InternalCalibration"
-           << "TOFCalibration"
-           << "FeatureFinderCentroided" << "FeatureFinderIsotopeWavelet" << "FeatureFinderMultiplex"
-           << "MassTraceExtractor" << "FeatureFinderMetabo"
-           << "FeatureFinderMRM"
-           << "IsobaricAnalyzer" << "SpectraFilterWindowMower"
-           << "SpectraFilterThresholdMower" << "SpectraFilterSqrtMower"
-           << "SpectraFilterParentPeakMower" << "SpectraFilterMarkerMower"
-           << "SpectraFilterScaler" << "SpectraFilterBernNorm"
-           << "SpectraFilterNLargest" << "SpectraFilterNormalizer";
-    }
-    else if (layer_type == LayerData::DT_FEATURE)
-    {
-      list << "FileFilter" << "FileConverter"
-           << "FileInfo" << "Decharger"
-           << "FeatureLinkerLabeled";
-    }
-    else if (layer_type == LayerData::DT_CONSENSUS)
-    {
-      list << "FileFilter" << "FileConverter"
-           << "FileInfo";
-    }
-    else if (layer_type == LayerData::DT_CHROMATOGRAM)
+    // Determine all available tools compatible with the layer_type
+    tool_map_ = {
```

Comment:
please use FileTypeList (feel free to extend the class), instead of string for more robustness:
https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/nightly/html/classOpenMS_1_1FileTypes_1_1FileTypeList.html

Reply:
I changed to map the reflect your suggestion. I had to change the following lines:
```suggestion
          // valid string is of format "*.TYPE" -> convert to substr "TYPE" in order to be comparable.
          const String& fileTypeStr = fileExtension.substr(2, fileExtension.size());
          const auto& iter = tool_map_.find(FileTypes::nameToType(subStr));
```
Each entry in of `valid_strings` is of format "*.TYPE". I sliced of the first two characters in order to be able to convert it to a `FileTypes::Type` with `FileTypes::nametoType` so it can be mapped. Is this what you had in mind? I don't see a way to get completely rid of using strings to determine the type.

Reply:
jep. All good.

This is unfortunately yet another case where our internal TOPP datastructures are not fully modeled in ParamXML.
In all TOPP tools we just use ParameterInformation, which contains the extension without `*.`.
But that is not related to this PR ...